internal/db/definition_validation.go

```diff
@@ -763,6 +727,84 @@ func validateTypeAndKindCompatible(
 	return nil
 }
 
+func validateEmbeddingAndKindCompatible(
```

Comment:
nitpick: Using 'vector embedding' terminology over just 'embedding' here and other places in this file.